develop/reference_workarounds.rst

```diff
@@ -5,20 +5,23 @@ SQL Support and Workarounds
 
 As Citus provides distributed functionality by extending PostgreSQL, it is compatible with PostgreSQL constructs. This means that users can use the tools and features that come with the rich and extensible PostgreSQL ecosystem for distributed tables created with Citus.
 
-Citus supports all SQL queries on distributed tables, with only these exceptions:
+Citus supports all SQL queries on distributed tables, with a few exceptions.
 
-* Correlated subqueries
-* `Recursive <https://www.postgresql.org/docs/current/static/queries-with.html#idm46428713247840>`_/`modifying <https://www.postgresql.org/docs/current/static/queries-with.html#QUERIES-WITH-MODIFYING>`_ CTEs
+Not yet supported:
+
+* `Recursive CTEs <https://www.postgresql.org/docs/current/static/queries-with.html#idm46428713247840>`_
 * `TABLESAMPLE <https://www.postgresql.org/docs/current/static/sql-select.html#SQL-FROM>`_
-* `SELECT … FOR UPDATE <https://www.postgresql.org/docs/current/static/sql-select.html#SQL-FOR-UPDATE-SHARE>`_
 * `Grouping sets <https://www.postgresql.org/docs/current/static/queries-table-expressions.html#QUERIES-GROUPING-SETS>`_
-* `Window functions <https://www.postgresql.org/docs/current/static/tutorial-window.html>`_ that do not include the distribution column in PARTITION BY
 
-Furthermore, in :ref:`mt_use_case` when queries are filtered by table :ref:`dist_column` to a single tenant then all SQL features work, including the ones above.
+Partially supported:
 
-To learn more about PostgreSQL and its features, you can visit the `PostgreSQL documentation <http://www.postgresql.org/docs/current/static/index.html>`_.
+* `SELECT … FOR UPDATE <https://www.postgresql.org/docs/current/static/sql-select.html#SQL-FOR-UPDATE-SHARE>`_ are supported for :ref:`router_executor` queries.
+* `Window functions <https://www.postgresql.org/docs/current/static/tutorial-window.html>`_ are supported when they include the distribution column in PARTITION BY.
+* Correlated subqueries are supported when the correlation is on the :ref:`dist_column` and the subqueries conform to subquery pushdown rules (e.g., grouping by the distribution column, with no LIMIT or LIMIT OFFSET clause).
+
+Furthermore, in :ref:`mt_use_case` when queries are filtered by table :ref:`dist_column` to a single tenant then all SQL features work, including the ones above.
```

Comment:
The following comment could be a personal opinion. But, I think instead of giving the following sentence in the end of the section:
```
Furthermore, in :ref:`mt_use_case` when queries are filtered by table :ref:`dist_column` to a single tenant then all SQL features work, including the ones above.
```

It might be more powerful to move the following even before: `Citus supports all SQL queries on distributed tables, with a few exceptions`. Such as:

```
To start with, in :ref:`mt_use_case` when queries are filtered by table :ref:`dist_column` to a single tenant then all SQL features work, including the ones below.
``` 

(You can consider changing `To start with` phrase to something more informal maybe?)

And, if we do the above, I'd suggest changing the title `Partially Supported ` to `Limitations for Parallel SQL Queries ` and list everything under that list.

That seems more accurate, because we don't have anything that should go under `Not yet supported:`, since now everything works for router queries. In the past, that probably made sense because some of things were not working for router queries such as tablesample and modifying ctes.

Reply:
Good point. I added a commit to structure it like you suggested. See what you think.